ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -138,17 +125,53 @@ export default ({
       });
 
       return reBatchedNodes;
+    },
+    [handleWorkflowUpdate, newEdgeCreation],
+  );
+
+  const handleCopy = useCallback(async () => {
+    const selected: { nodeIds: string[]; edges: Edge[] } | undefined = {
+      nodeIds: nodes.filter((n) => n.selected).map((n) => n.id),
+      edges: edges.filter((e) => e.selected),
+    };
+
+    const selectedNodes = nodes.filter((n) => selected.nodeIds.includes(n.id));
```

Comment:
yeah don't need this. Just make the above nodes and edges, not nodeIds and you are good to go 👍🏻